File: SwiftProject/SwiftProject/K06.swift
//
//  K06.swift
//  SwiftProject
//
//  Created by 周哲輝 on 2019/05/05.
//  Copyright © 2019 周哲輝. All rights reserved.
//

import Foundation

public class K06 : IExecutable {
    public override func Question1() {
        var ages = [Int]()
        
        for i in 1...5 {
            print("\(i)人目の年齢を入力してください＞", terminator: "")
            ages.append(Int(readLine()!)!)
        }
        
        let average = ages.reduce(0, +) / ages.count
        print("\(ages.count)人の平均年齢は\(average)です。")
    }
    
    public override func Question2() {
        for i in 1...9 {
            for _ in 1...i {
                print("*", terminator: "")
            }
            print()
        }
        print()
        
        for i in stride(from: 9, to: 0, by: -1) {
            for _ in 1...i {
                print("*", terminator: "")
            }
            print()
        }
        print()
        
        for i in stride(from: 9, to: 0, by: -1) {
            for _ in stride(from: 0, to: i - 1, by: 1) {
                print(" ", terminator: "")
            }
            
            for _ in stride(from: 9, to: i - 1, by: -1) {
                print("*", terminator: "")
            }
            print()
        }
    }
    
    public override func Question3() {
        let total = 370
        var count = 0
        
        for i in 0...(total / 100) {
            for j in 0...(total / 50) {
                for k in 0...(total / 10) {
                    if 100 * i + 50 * j + 10 * k == total {
                        print("10円の硬貨\(k)枚　50円の硬貨\(j)枚　100円の硬貨\(i)枚")
                        count += 1
                    }
                }
            }
        }
        
        print("以上\(count)通りを発見しました。")
    }
    
    public override func Question4() {
        print("\t|\t", terminator: "")
        for i in 1...9 {
            print("\(i)\t", terminator: "")
        }
        print()
        
        for _ in 0...75 {
            print("-", terminator: "")
        }
        print()
        
        for i in 1...9 {
            print("\(i)\t|\t", terminator: "")
            for j in 1...9 {
                print("\(i * j)\t", terminator: "")
            }
            print()
        }
    }
}

File: Scala/src/main/scala/PrimaryClasses/K06.scala
package PrimaryClasses
import IExecutable.IExecutable
import scala.collection.mutable


class K06 extends IExecutable {
  override def Question1(): Unit = {
    var ages = mutable.MutableList[Float]()
    for (i <- 0 to 4) {
      printf("%d人目の年齢を入力して下さい：", i + 1)
      ages += scala.io.StdIn.readFloat()
    }

    val average = ages.sum / ages.length.toFloat
    printf("%d人の平均年齢は%f", ages.length, average)
  }

  override def Question2(): Unit = {
    for (i <- 1 to 9) {
      for (j <- 1 to i) {
        print("*")
      }
      println()
    }
    println()

    for (i <- 9 to 1 by -1) {
      for (j <- 1 to i) {
        print("*")
      }
      println()
    }
    println()

    for (i <- 9 to 1 by -1) {
      for (j <- 1 until i) {
        print(" ")
      }
      for (k <- 9 to i by -1) {
        print("*")
      }
      println()
    }
  }

  override def Question3(): Unit = {
    val total = 370
    var count = 0
    for {
      i <- 0 to total / 100
      j <- 0 to total / 50
      k <- 0 to total / 10
    } {
      if (100 * i + 50 * j + 10 * k == total) {
        printf("10円の硬貨%d枚 50円の硬貨%d枚 100円の硬貨%d枚\n", k, j, i)
        count += 1
      }
    }

    printf("\n以上%d通りを発見しました。", count)
  }

  override def Question4(): Unit = {
    print("\t|\t")
    for (i <- 1 to 9) {
      printf("%d\t", i)
    }
    println("")

    for (i <- 1 to 75) {
      print("-")
    }
    println("")

    for (i <- 1 to 9) {
      printf("%d\t|\t", i)
      for (j <- 1 to 9) {
        printf("%d\t", i * j)
      }
      println("")
    }
  }
}
